src/Message/ReplaceCardRequest.php

```diff
@@ -0,0 +1,140 @@
+            'BillingAddress' => array(
+                'NameOnCard' => $this->getNameOnCard(),
+                'AddressLineOne' => $card->getBillingAddress1(),
+                'City' => $card->getBillingCity(),
+                'State' => $card->getBillingState(),
+                'ZipCode' => substr($card->getBillingPostcode(), 0, 5),
+                'Phone' => preg_replace("/[^0-9]/", '', $card->getBillingPhone()),
+            ),
+            'AccountUsePreferenceType' => 'MultiUse'
+        );
+
+        $data['customer'] = array(
+            'FirstName' => $card->getFirstName(),
+            'LastName' => $card->getLastName(),
+            'AdressLineOne' => $card->getBillingAddress1(),
+            'City' => $card->getBillingCity(),
+            'State' => $card->getBillingState(),
+            'Zip' => substr($card->getBillingPostcode(), 0, 5),
+            'CustomerReferenceCode' => $this->getCustomerReferenceCode(),
+        );
+
+        $data['merchantPayeeCode'] = $this->getMerchantPayeeCode();
+
+        return $data;
+    }
+
+    /**
+     * Get authentication strings
+     *
+     * @return array
+     */
+    private function getAuthenticationParams()
+    {
+        return array(
+            'PvLogin' => $this->getPvLogin(),
+            'PvPassword' => $this->getPvPassword(),
+            'PvAPIKey' => $this->getPvAPIKey(),
+            'PvToken' => $this->getPvToken(),
+        );
+    }
+
+    /**
+     * Send the request with specified data
+     *
+     * @param  mixed $data The data to send
+     * @return ResponseInterface
+     */
+    public function sendData($data)
+    {
+        if (true === $this->getStubMode()) {
+            $response = $this->getFakeResponse($data);
+            return $this->response = new FakeResponse($this, $response);
+        }
+
+        if (!$this->soap) {
+            $this->soap = new \SoapClient($this->getWsdl(), array('trace' => $this->getTestMode()));
+        }
+
+        $response = call_user_func_array(array($this->soap, 'ReplaceCreditCardAccount'), array($data));
+
+        return $this->response = new Response($this, $response);
+    }
+
+    /**
+     * Get fake response specific to this request
+     *
+     * @param  mixed $data The data that would otherwise be sent
+     * @return object
+     */
+    public function getFakeResponse($data)
+    {
+        $dateString = date('YmdHis');
+        return (object) [
+            'ReplaceCreditCardAccountResult' => [
+                'Responses' => [
+                    'Response' => [
+                        'ResponseCode' => '1000',
+                        'ErrorMessage' => ''
+                    ]
+                ],
+                'TimeReceived' => date('n/j/Y g:i:s A'),
+                'CustomerReferenceCode' => 'CU' . $dateString,
+                'CreditCardAccountToken' => [
+                    'ReferenceID' => $data['referenceID'],
```

Comment:
The real response would end up returning a new ReferenceID. In other fake responses, like CreateCardRequest for example, we're using a date-time string as the fake ID.

Reply:
Like this?
https://github.com/RTONational/omnipay-paymentvision/pull/18/commits/c9f742b36171d36ad5d0099d629efe7165068860

Reply:
Exactly.